libbeat/filter/condition.go

```diff
@@ -0,0 +1,330 @@
+
+	for field, value := range equals {
+
+		i, err := strconv.Atoi(value)
+		if err == nil {
+			c.Equals[field] = EqualsValue{Int: i}
+		} else {
+			c.Equals[field] = EqualsValue{Str: value}
+		}
+	}
+	return nil
+}
+
+func (c *Condition) AddContains(contains map[string]string) error {
+
+	c.Contains = contains
+	return nil
+}
+
+func (c *Condition) AddRegexp(r map[string]string) error {
+
+	for field, value := range r {
+		reg, err := regexp.CompilePOSIX(value)
+		if err != nil {
+			return err
+		}
+		c.Regexp[field] = reg
+	}
+	return nil
+}
+
+func (c *Condition) AddRange(config map[string]RangeValue) error {
+
+	for field, rangeValue := range config {
+		c.Range[field] = rangeValue
+	}
+	return nil
+}
+
+func (c *Condition) Check(event common.MapStr) bool {
+
+	if !c.CheckEquals(event) {
+		return false
+	}
+	if !c.CheckContains(event) {
+		return false
+	}
+	if !c.CheckRegexp(event) {
+		return false
+	}
+	if !c.CheckRange(event) {
+		return false
+	}
+
+	return true
+}
+
+func (c *Condition) CheckEquals(event common.MapStr) bool {
+
+	for field, equalValue := range c.Equals {
+
+		value, err := event.GetValue(field)
+		if err != nil {
+			logp.Err("unavailable field %s: %v", field, err)
+			return false
+		}
+
+		switch value.(type) {
+		case uint8, uint16, uint32, uint64, int8, int16, int32, int64, int, uint:
+			return value == equalValue.Int
+		case string:
+			return value == equalValue.Str
+		default:
+			logp.Err("unexpected type %T in equals condition as it accepts only integers and strings. ", value)
+			return false
+		}
+
+	}
+
+	return true
+
+}
+
+func (c *Condition) CheckContains(event common.MapStr) bool {
+
+	for field, equalValue := range c.Contains {
+
+		value, err := event.GetValue(field)
+		if err != nil {
+			logp.Err("unavailable field %s: %v", field, err)
+			return false
+		}
+
+		switch value.(type) {
+		case string:
+			return strings.Contains(value.(string), equalValue)
+		default:
+			logp.Err("unexpected type %T in contains condition as it accepts only strings. ", value)
+			return false
+		}
+
+	}
+
+	return true
+
+}
+
+func (c *Condition) CheckRegexp(event common.MapStr) bool {
+
+	for field, equalValue := range c.Regexp {
+
+		value, err := event.GetValue(field)
+		if err != nil {
+			logp.Err("unavailable field %s: %v", field, err)
+			return false
+		}
+
+		switch value.(type) {
+		case string:
+			if !equalValue.MatchString(value.(string)) {
+				return false
+			}
+		default:
+			logp.Err("unexpected type %T in regexp condition as it accepts only strings. ", value)
+			return false
+		}
+
+	}
+
+	return true
+
+}
+
+func (c *Condition) CheckRange(event common.MapStr) bool {
+
+	for field, rangeValue := range c.Range {
+
+		value, err := event.GetValue(field)
+		if err != nil {
+			logp.Err("unavailable field %s: %v", field, err)
+			return false
+		}
+
+		switch value.(type) {
+		case int, int8, int16, int32, int64:
+			int_value := reflect.ValueOf(value).Int()
```

Comment:
@urso Thanks! I already changed it with reflect. Please let me know if you find any other issues.